PluginsAndFeatures/azure-toolkit-for-intellij/rider/ReSharper.Azure/src/Azure.FSharp/FunctionApp/CodeCompletion/Rules/FSharpTimerTriggerCronArgumentsProvider.cs

```diff
@@ -0,0 +1,145 @@
+
+namespace JetBrains.ReSharper.Azure.FSharp.FunctionApp.CodeCompletion.Rules
+{
+    [Language(typeof(FSharpLanguage))]
+    public class FSharpTimerTriggerCronArgumentsProvider : ItemsProviderOfSpecificContext<FSharpCodeCompletionContext>
+    {
+        // TODO: Refactor so C# and F# share this
+        private readonly struct CronSuggestion
+        {
+            public readonly string Expression;
+            public readonly string Description;
+
+            public CronSuggestion(string expression, string description)
+            {
+                Expression = expression;
+                Description = description;
+            }
+        }
+
+        // TODO: Refactor so C# and F# share this
+        private readonly CronSuggestion[] CronSuggestions =
+        {
+            new CronSuggestion("* * * * * *", "Every second"),
+            new CronSuggestion("0 * * * * *", "Every minute"),
+            new CronSuggestion("0 */5 * * * *", "Every 5 minutes"),
+            new CronSuggestion("0 0 * * * *", "Every hour"),
+            new CronSuggestion("0 0 */6 * * *", "Every 6 hours at minute 0"),
+            new CronSuggestion("0 0 8-18 * * *", "Every hour between 08:00 AM and 06:59 PM"),
+            new CronSuggestion("0 0 0 * * *", "At 12:00 AM"),
+            new CronSuggestion("0 0 10 * * *", "At 10:00 AM"),
+            new CronSuggestion("0 0 * * * 1-5", "Every hour, Monday through Friday"),
+            new CronSuggestion("0 0 0 * * 0", "At 12:00 AM, only on Sunday"),
+            new CronSuggestion("0 0 9 * * Mon", "At 09:00 AM, only on Monday"),
+            new CronSuggestion("0 0 0 1 * *", "At 12:00 AM, on day 1 of the month"),
+            new CronSuggestion("0 0 0 1 1 *", "At 12:00 AM, on day 1 of the month, only in January"),
+            new CronSuggestion("0 0 * * * Sun", "Every hour, only on Sunday"),
+            new CronSuggestion("0 0 0 * * Sat,Sun", "At 12:00 AM, only on Saturday and Sunday"),
+            new CronSuggestion("0 0 0 * * 6,0", "At 12:00 AM, only on Saturday and Sunday"),
+            new CronSuggestion("0 0 0 1-7 * Sun", "At 12:00 AM, between day 1 and 7 of the month, only on Sunday"),
+            new CronSuggestion("11 5 23 * * *", "At 11:05:11 PM"),
+            new CronSuggestion("*/15 * * * * *", "Every 15 seconds"),
+            new CronSuggestion("0 30 9 * Jan Mon", "At 09:30 AM, only on Monday, only in January")
+        };
+
+        protected override bool IsAvailable(FSharpCodeCompletionContext context)
+        {
+            return IsStringLiteral(context) &&
+                   IsTimerTriggerAnnotation(context) &&
+                   context.BasicContext.CodeCompletionType == CodeCompletionType.BasicCompletion;
+        }
+
+        protected override bool AddLookupItems(FSharpCodeCompletionContext context, IItemsCollector collector)
+        {
+            foreach (var cronSuggestion in CronSuggestions)
+            {
+                var token = context.TokenAtCaret as ITokenNode;
```

Comment:
@auduchinok Seems completion can only be triggered right before a string, so `{caret}"....."` triggers completion, `"...{caret}...."` does not. This completion provider now assumes the quotes are always included. This means that on L98 below, I have to provide completion items with `"` ?

(I could be misunderstanding the PSI as well)